actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -231,6 +231,9 @@ class Parameters
     #   params.permitted?  # => true
     #   Person.new(params) # => #<Person id: nil, name: "Francesco">
     def initialize(parameters = {})
+      unless parameters.respond_to?(:with_indifferent_access)
```

Comment:
I agree with @matthewd, it's quite expensive and also this implementation hides the errors can occur inside of the `with_indifferent_access` method. If we do something wrong inside of this method which raises `NoMethodError` then we say `ArgumentError` which creates much more confusion than what it does now.